src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incidents/IncidentsMigrationService.kt

```diff
@@ -70,33 +71,68 @@ class IncidentsMigrationService(
     incidentsMappingService.findByNomisId(nomisIncidentId)
       ?.run {
         log.info("Will not migrate the nomis incident=$nomisIncidentId since it was already mapped to DPS incident ${this.dpsIncidentId} during migration ${this.label}")
-      } ?: runCatching {
-      val nomisIncidentResponse = nomisApiService.getIncident(nomisIncidentId)
-      incidentsService.upsertIncident(nomisIncidentResponse.toMigrateUpsertNomisIncident()).also {
-        createIncidentMapping(
-          dpsIncidentId = it.id.toString(),
-          nomisIncidentId = nomisIncidentId,
-          context = context,
+      }
+      ?: try {
+        val nomisIncidentResponse = nomisApiService.getIncident(nomisIncidentId)
+        incidentsService.upsertIncident(nomisIncidentResponse.toMigrateUpsertNomisIncident()).also {
+          createIncidentMapping(
+            dpsIncidentId = it.id.toString(),
+            nomisIncidentId = nomisIncidentId,
+            context = context,
+          )
+          telemetryClient.trackEvent(
+            "${MigrationType.INCIDENTS.telemetryName}-migration-entity-migrated",
+            mapOf(
+              "nomisIncidentId" to nomisIncidentId,
+              "dpsIncidentId" to it.id,
+              "migrationId" to migrationId,
+            ),
+          )
+        }
+      } catch (e: WebClientResponseException.Conflict) {
+        // We have a conflict - this should only ever happen if the incident was stored in DPS, but we didn't receive a response in time
+        // so is never added to the incidents mapping table
```

Comment:
Is there a code to check too? e.g. locations api sends errorCode == 107

Reply:
yeah I saw you had one of those - I have just checked - they have one (302) but I'm not sure if I need it - will take a look